R/tm_missing_data.R

```diff
@@ -337,12 +349,17 @@ srv_missing_data <- function(id, data, reporter, filter_panel_api, dataname, plo
     data_r <- data[[dataname]]
 
     data_keys <- reactive(get_join_keys(data)$get(dataname)[[dataname]])
+
     data_parent_keys <- reactive({
-      keys <- get_join_keys(data)$get(dataname)
-      if ("ADSL" %in% names(keys)) {
-        keys[["ADSL"]]
+      if (length(parent_dataname) > 0 && parent_dataname %in% names(data)) {
+        keys <- get_join_keys(data)$get(dataname)
+        if (parent_dataname %in% names(keys)) {
+          keys[[parent_dataname]]
+        } else {
+          keys[[dataname]]
+        }
       } else {
-        keys[[dataname]]
+        NULL
```

Comment:
NULL should be ok here as the value is not used when if_patients_plot is not there which needs parent_dataname